libphal/phal_sbe.C

```diff
@@ -370,5 +392,215 @@ void threadStopProc(struct pdbg_target *proc)
+ * not bad.
+ *
+ * If both primary and secondary seeprom's are bad throw exception
+ * to the caller so that the processor can be deconfigured/guarded.
+ *
+ * @param[in] proc processor target to operate on
+ * @return throws exception if not able to switch or for any
+ *  attribute read/write error.
+ */
+void reiplUsingBKPBootSeeprom(struct pdbg_target *proc)
+{
+	log(level::INFO, "SBERC:reiplUsingBKPBootSeeprom on %s",
+	    pdbg_target_path(proc));
+}
+
+/**
+ * @brief ReIPL using backup measurement seeprom
+ *
+ * Current sbe seeprom/sbe measurement seeprom code used by the
+ * the processor could be faulty due to which chip-op could fail.
+ *
+ * p10_extract_sbe_rc HWP is run to identify the recovery action
+ * for the SBE seeprom failure.
+ *
+ * This method is to cater to ReIPL using backup meas seeprom
+ * recovery action specified by p10_extract_sbe_rc. This could
+ * be because the current side of the seeprom is not recoverable.
+ *
+ * Check the currently used measurement seeprom side used and mark
+ * it as bad and then select the backup measurement seeprom if the
+ * same is not bad.
+ *
+ * If both primary and secondary seeprom's are bad throw exception
+ * to the caller so that the processor can be deconfigured/guarded.
+ *
+ * @param[in] proc processor target to operate on
+
+ * @return throws exception if not able to switch or for any
+ * 		attribute read/write error.
+ */
+void reiplUsingBKPMeasurementSeeprom(struct pdbg_target *proc)
+{
+	log(level::INFO, "SBERC:reiplUsingBKPMeasurementSeeprom on %s",
+	    pdbg_target_path(proc));
+}
+
+/**
+ * @brief Handle SBE error and perform recovery action
+ *
+ * For any SBE access/chip-op failure SBE FFDC is collected,
+ * If SBE FFDC collection fails with a Timeout error, then
+ * we assume SBE is in a bad state. Execute p10_extract_sbe_rc
+ * HWP to determine the SBE failure reason and the recovery
+ * action to be done by BMC for the failure.
+ *
+ * Routine should be called as part of SBE error handling
+ *
+ * This method puts the SBE state as FAILED in device tree
+ *
+ * @param[in] proc Non NULL proc Target handle.
+ * @return Exception thrown on failure
+ */
+void sbeErrorHandler(struct pdbg_target *proc)
+{
+	try {
+		assert(proc != NULL);
+		log(level::INFO, "SBERC: sbeErrorHandler on %s",
+		    pdbg_target_path(proc));
+
+		auto sbeState = getState(proc);
+
+		if (sbeState == SBE_STATE_DEBUG_MODE ||
+		    sbeState == SBE_STATE_FAILED) {
+			// The SBE on this processor is not in a state
+			// where chipOps can be issued. SBE Error handling
+			// will be skipped.
+			log(level::ERROR,
+			    "SBERC: sbe for proc %s is not in DEBUG/FAILED "
+			    "state %d",
+			    pdbg_target_path(proc), sbeState);
+			throw sbeError_t(exception::SBE_INVALID_STATE);
+		}
+
+		// set state to debug mode to extract sbe rc
+		setState(proc, SBE_STATE_DEBUG_MODE);
+
+		// Execute SBE extract rc to set up sdb bit for pibmem dump to
```

Comment:
are we missing something based on this comment? is it unwanted?

Reply:
unwanterd